### ISSUE TITLE
[MPS] Add support for multi-dimensional all / any reductions

### ISSUE THREAD

**Issue by AneeshGidda** [pytorch/pytorch#176826]
The MPS backend does not currently provide a dedicated native implementation for multi-dimensional reductions in `torch.all` and `torch.any` when `dim` is provided as multiple dimensions.

Example:

```python
x = torch.ones(2, 3, 4, device="mps", dtype=torch.bool)

torch.all(x, dim=[0, 1])
torch.any(x, dim=[1, 2])
```

These operations currently work through the composite implementation, which performs reductions sequentially across dimensions. While each individual reduction may use an existing MPS kernel, there is no native MPS implementation that reduces across multiple dimensions in a single operation.

Adding a native implementation for this case would improve backend feature parity with CPU and CUDA and could allow the reduction to be performed more efficiently on MPS.

I have a working implementation with tests and would be happy to open a PR for this.

cc @jerryzh168 @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk

**Comment by kurtamohler** [comment]
This has been fixed by #180752
